bsp/worker/src/mill/bsp/worker/Utils.scala

```diff
@@ -46,6 +58,75 @@ private object Utils {
     else StatusCode.OK
   }
 
+  def makeBuildTarget(
+      id: BuildTargetIdentifier,
+      depsIds: Seq[BuildTargetIdentifier],
+      bt: BspBuildTarget,
+      data: Option[(String, Object)]
+  ): BuildTarget = {
+    val buildTarget = new BuildTarget(
+      id,
+      bt.tags.asJava,
+      bt.languageIds.asJava,
+      depsIds.asJava,
+      new BuildTargetCapabilities().tap { it =>
+        it.setCanCompile(bt.canCompile)
+        it.setCanTest(bt.canTest)
+        it.setCanRun(bt.canRun)
+        it.setCanDebug(bt.canDebug)
+      }
+    )
+
+    bt.displayName.foreach(buildTarget.setDisplayName)
+    bt.baseDirectory.foreach(p => buildTarget.setBaseDirectory(sanitizeUri(p)))
+
+    for ((dataKind, data) <- data) {
+      buildTarget.setDataKind(dataKind)
+      buildTarget.setData(data)
+    }
+    buildTarget
+  }
+
+  def outputPaths(
+      outDir: os.Path,
+      buildTargetBaseDir: os.Path,
+      topLevelProjectRoot: os.Path
+  ): Seq[OutputPathItem] = {
+    val output = new OutputPathItem(
+      // Spec says, a directory must end with a forward slash
+      sanitizeUri(outDir) + "/",
+      OutputPathItemKind.DIRECTORY
+    )
+
+    def ignore(path: os.Path): Boolean = {
+      path.last.startsWith(".") ||
+      path.endsWith(os.RelPath("out")) ||
+      path.endsWith(os.RelPath("target")) ||
+      path.endsWith(os.RelPath("docs")) ||
```

Comment:
Why are we ignoring `docs/`? I'm not aware of such a folder name being treated specially by any build tool

Reply:
Hmm , I assumed that we don't want to index this ,however I might have been wrong. I don't mind removing docs exclusion (and potentially adding anything that comes to your mind)